runtime/src/bank_utils.rs

```diff
@@ -59,3 +62,55 @@ pub fn find_and_send_votes(
             });
     }
 }
+
+/// Logs the measurement values
+pub fn submit_loaded_programs_stats(stats: &Stats, slot: Slot) {
+    let hits = stats.hits.load(Ordering::Relaxed);
+    let misses = stats.misses.load(Ordering::Relaxed);
+    let evictions: u64 = stats.evictions.values().sum();
+    let reloads = stats.reloads.load(Ordering::Relaxed);
+    let insertions = stats.insertions.load(Ordering::Relaxed);
+    let lost_insertions = stats.lost_insertions.load(Ordering::Relaxed);
+    let replacements = stats.replacements.load(Ordering::Relaxed);
+    let one_hit_wonders = stats.one_hit_wonders.load(Ordering::Relaxed);
+    let prunes_orphan = stats.prunes_orphan.load(Ordering::Relaxed);
+    let prunes_environment = stats.prunes_environment.load(Ordering::Relaxed);
+    let empty_entries = stats.empty_entries.load(Ordering::Relaxed);
+    datapoint_info!(
+        "loaded-programs-cache-stats",
+        ("slot", slot, i64),
+        ("hits", hits, i64),
+        ("misses", misses, i64),
+        ("evictions", evictions, i64),
+        ("reloads", reloads, i64),
+        ("insertions", insertions, i64),
+        ("lost_insertions", lost_insertions, i64),
+        ("replace_entry", replacements, i64),
+        ("one_hit_wonders", one_hit_wonders, i64),
+        ("prunes_orphan", prunes_orphan, i64),
+        ("prunes_environment", prunes_environment, i64),
+        ("empty_entries", empty_entries, i64),
+    );
+    debug!(
+            "Loaded Programs Cache Stats -- Hits: {}, Misses: {}, Evictions: {}, Reloads: {}, Insertions: {} Lost-Insertions: {}, Replacements: {}, One-Hit-Wonders: {}, Prunes-Orphan: {}, Prunes-Environment: {}, Empty: {}",
+            hits, misses, evictions, reloads, insertions, lost_insertions, replacements, one_hit_wonders, prunes_orphan, prunes_environment, empty_entries
+        );
```

Comment:
Although `datapoint_*` macro calls don't emit the call site of the metric (it's labeled with `solana_metrics::metrics`), unfortunately `debug!`, `trace!` and the other log macros will in fact emit `solana_program_runtime::loaded_programs`, which would now be changed to `solana_runtime::bank`.

Maybe just hoist the `datapoint_info!` bit out to the runtime, and leave the logs in `LoadedProgramStats::log(&self)` and just call `stats.log()` in the runtime function `submit_loaded_program_metrics`?

Reply:
Done, had to make a choice between a new struct or a method with 11 parameters